homeassistant/components/recorder/__init__.py

```diff
@@ -291,7 +304,8 @@ def __init__(
         self.get_session = None
         self._completed_database_setup = None
         self._event_listener = None
-
+        self.async_migration_event = asyncio.Event()
```

Comment:
Hmm.. there is a race here since they can call stop before startup is finished.  We probably need to set this as soon as we hit the point of checking the future.

Reply:
A user would have to request a restart between when the `hass_started.result` call finishes and  the variable is swapped.  If we set the var before we check the result then they can't stop the  instance because something is blocking startup which probably won't work anyways.  Probably best to set it before since they upgraded intentionally and we are already at the point of no return.

Reply:
Yes